core/downsampling/src/main/java/uk/ac/cam/cares/Downsampling.java

```diff
@@ -0,0 +1,220 @@
+package uk.ac.cam.cares;
+
+import uk.ac.cam.cares.jps.base.timeseries.TimeSeries;
+
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+public class Downsampling {
+
+    /**
+     * @param ts Input timeseries
+     * @param resolution Input resolution - interval to be aggregated
+     * @param type Input type
+     * @return
+     * @throws Exception
+     */
+
+    public static TimeSeries aggregation(TimeSeries ts, Long resolution, int type) throws Exception {
+        //Parsing timseries into list of list values and time list;
+        List TSDataIRIS = ts.getDataIRIs();
+        List<List<Double>> TSlolValues = new ArrayList<>();
+
+        for (Object TSDataIRI : TSDataIRIS) {
+            TSlolValues.add(ts.getValues(TSDataIRI.toString()));
+        }
+        List<OffsetDateTime> originalTimeList = ts.getTimes();
+
+        List ResampledList;
+        ResampledList = aggregationMethod(originalTimeList, TSlolValues, resolution,type);
+
+        TimeSeries resampledTS = new TimeSeries((List) ResampledList.get(0), (List<String>) TSDataIRIS, (List<List<?>>) ResampledList.get(1));
+
+        return resampledTS;
+    }
+
+    public static List aggregationMethod(List<OffsetDateTime> originalTimeList, List<List<Double>> originalValueLists, long intervalInSeconds, int type) throws Exception {
+        List<List<Double>> resampledValueLists = new ArrayList<>();
+
+        //Initiliaze size of resampledValueLists for iterator purpose
+        for (int i = 0; i < originalValueLists.size(); i++) {
+            resampledValueLists.add(new ArrayList<>());
+        }
```

Comment:
this for loop can be removed and resampledValueLists can be initialized in one line using Collections.nCopies